src/Neo/Network/P2P/Payloads/NetworkAddressWithTime.cs

```diff
@@ -75,7 +75,10 @@ void ISerializable.Deserialize(ref MemoryReader reader)
             Capabilities = new NodeCapability[reader.ReadVarInt(VersionPayload.MaxCapabilities)];
             for (int x = 0, max = Capabilities.Length; x < max; x++)
                 Capabilities[x] = NodeCapability.DeserializeFrom(ref reader);
-            if (Capabilities.Select(p => p.Type).Distinct().Count() != Capabilities.Length)
+            // Verify that no duplicating capabilities are included. Unknown capabilities are not
+            // taken into account but still preserved to be able to share through the network.
+            var capabilities = Capabilities.Where(c => c is not UnknownCapability);
+            if (capabilities.Select(p => p.Type).Distinct().Count() != capabilities.Count())
```

Comment:
why u comparing `capabilities` with `capabilities` and not `Capabilities`? This logic isn't the same or even close. I thought we were allowing one unknown capability?

Reply:
That's intentional. Peer can send a number of unknown capabilities with different or same types, it's OK (since we have no idea what these capabilities are about and can't say whether it's a bug or a feature). We're ensuring uniqueness for types we know of.